### ISSUE TITLE
photo sharing SDK for mobile and other devices

### ISSUE THREAD

**Issue by vkkhare** [photoprism/photoprism#1384]
I would like to know if PhotoPrism is interested in supporting distributed cloud over users' devices.

The users do not need to set up their own servers, just install the respective mobile app/desktop clients. They all will sync in real-time with each other. The storage itself is distributed over the local network.

**Comment by alexislefebvre** [comment]
Is there an existing protocol about these features? It looke like it would work without ProtoPrism, how does it relate to this project?

**Comment by vkkhare** [comment]
There are few benefits to photo prism from my perspective:
 - Currently, Photoprism runs a server in go and clients connect to it via a progressive web app. I also found a native in development. 
 - Those apps can do on-device tagging to reduce the computational load on the central server and in some cases even remove the need completely. Privacy-preserving ML stack like www.openmined.org can help us here
 - The apps could also support local indexing so that even when the user doesn't have access to the internet he can still view the files distributed across his devices with proper linking (say his phone, laptop, and iPad connected by BlueTooth). I would like to make this happen locally distributed over individual devices. 
 - WebDAV clients can display the photographs (over intranet only though), but the indexing (referencing) and ML still need to be done by PhotoPrism.

**Comment by alexislefebvre** [comment]
Is it something you imagine in the [mobile app](https://github.com/photoprism/photoprism-mobile)?

**Comment by vkkhare** [comment]
yup @alexislefebvre, though I remember, there is a progressive web app as well somewhere?

**Comment by alexislefebvre** [comment]
Yes, every PhotoPrism provide a PWA: https://github.com/photoprism/photoprism/blob/b90599ddfcb75f2a570d0a50c2f175d2b344c9eb/frontend/package.json#L4

**Comment by lastzero** [comment]
You can use our existing REST API with external apps (mobile or server based). While we want to provide an official SDK at some point in the future, you don't need one as our API doesn't use proprietary technology.

There are also existing client libs like https://docs.photoprism.org/developer-guide/clients/go/. If you develop something, we're happy to add a link to our docs.
